r/src/table.cpp

```diff
@@ -88,7 +88,7 @@ std::shared_ptr<arrow::Table> Table__Slice2(const std::shared_ptr<arrow::Table>&
 
 // [[arrow::export]]
 bool Table__Equals(const std::shared_ptr<arrow::Table>& lhs,
-                   const std::shared_ptr<arrow::Table>& rhs) {
+                   const std::shared_ptr<arrow::Table>& rhs, bool check_metadata) {
```

Comment:
I think you need to pass `check_metadata` on the next line.

Reply:
Uh, what do you mean? All CI passed (except for an unrelated AppVeyor issue).

Reply:
Exactly my point about the test. We're accepting this new argument but not passing it to the C++ function, so this is not really implemented, and we would have known that if we had a test.

Reply:
Ahah, sorry, I had misunderstood you. Hmm, can I let you open a JIRA and submit a PR?

Reply:
https://issues.apache.org/jira/browse/ARROW-7786